### PR TITLE
Feat update total grants metric

### PR DESCRIPTION
**Pull request by hcdeng**
This addresses https://airtable.com/tblk2ToSNYzNVza5f/viwrmqbUpwcUGA2OM/rec5IYo3f72Qern8H?blocks=bip4xBsHtRAeV2Zb

- adds "total grants matching search criteria" metric to dashboard
- links the "total grants matching criteria" card on dashboard to grants page
- adds a `helpers` file for shared logic across db calls

~Working on tests, but wanted to get this out for early feedback.~ Tests added! You can run `yarn run test:db` from the /server directory. Let me know your thoughts!

image

### CODE REVIEW

**Comment by Michael-Stanford** [comment]
Looks good to me!

**Review by smattingly — APPROVED** [review]
I can't improve on this. I learned some about Vuex by reviewing it.

File: packages/server/src/db/index.js [hunk 1]
```diff
@@ -244,8 +249,8 @@ async function getGrant({ grantId }) {
     return results[0];
 }
 
-async function getTotalGrants() {
-    const rows = await knex(TABLES.grants).count();
+async function getTotalGrants({ filters } = {}) {
+    const rows = await knex(TABLES.grants).modify(helpers.whereFiltersMatch, filters).count();
```
Comment (Rapol) [inline packages/server/src/db/index.js:253]: TIL, didnt know about modify

**Review by Rapol — APPROVED** [review]
Nice job!

**Comment by hcdeng** [comment]
@Rapol Fixed merge conflicts. Decided to call the keywords/eligibilityCodes "agencyCriteria" to avoid interfering too much with your "assigned/interested by user" filters. I also added some db tests inspired by @lhitchon 's work in the cares-reporter. Let me know what you think!

**Pull request merged**